plugins/bil/bil_main.ml

```diff
@@ -34,44 +33,46 @@ let eliminate_dead_code bil =
       | st -> loop (st :: acc) bil in
   loop [] bil
 
-let apply bil =
-  List.fold ~init:bil ~f:(fun bil f -> f bil)
-
-let norml level =
-  if level = 0 then ident
-  else
-    Stmt.normalize ~keep_ites:false ~normalize_exp:false
+let () =
+  Bil.register_pass "normalization" @@
+  Stmt.normalize ~keep_ites:false ~normalize_exp:false
 
-let simpl level =
-  if level = 0 then ident
-  else
-    Bil.fixpoint
-      (fun bil -> Bil.fold_consts bil |> Bil_const_propagation.run |> eliminate_dead_code)
+let () =
+  Bil.register_pass "optimization" @@
+  Bil.fixpoint
```

Comment:
remove fixed point from here. You shall rely on the framework's fixed point instead.